src/modules/powerrename/lib/PowerRenameManager.cpp

```diff
@@ -823,192 +831,203 @@ DWORD WINAPI CPowerRenameManager::s_regexWorkerThread(_In_ void* pv)
+                            }
+                        }
+                        else
+                        {
+                            StringCchCopy(resultName, ARRAYSIZE(resultName), newName);
                         }
                     }
+
+                    wchar_t trimmedName[MAX_PATH] = { 0 };
+                    if (newNameToUse != nullptr)
+                    {
+                        winrt::check_hresult(GetTrimmedFileName(trimmedName, ARRAYSIZE(trimmedName), newNameToUse));
+                        newNameToUse = trimmedName;
+                    }
+
+                    wchar_t transformedName[MAX_PATH] = { 0 };
+                    if (newNameToUse != nullptr && (flags & Uppercase || flags & Lowercase || flags & Titlecase))
+                    {
+                        winrt::check_hresult(GetTransformedFileName(transformedName, ARRAYSIZE(transformedName), newNameToUse, flags));
+                        newNameToUse = transformedName;
+                    }
+
+                    // No change from originalName so set newName to
+                    // null so we clear it from our UI as well.
+                    if (lstrcmp(originalName, newNameToUse) == 0)
+                    {
+                        newNameToUse = nullptr;
+                    }
+
+                    wchar_t uniqueName[MAX_PATH] = { 0 };
+                    if (newNameToUse != nullptr && (flags & EnumerateItems))
+                    {
+                        unsigned long countUsed = 0;
+                        if (GetEnumeratedFileName(uniqueName, ARRAYSIZE(uniqueName), newNameToUse, nullptr, itemEnumIndex, &countUsed))
+                        {
+                            newNameToUse = uniqueName;
+                        }
+                        itemEnumIndex++;
+                    }
+
+                    winrt::check_hresult(spItem->PutNewName(newNameToUse));
+
+                    // Was there a change?
+                    if (lstrcmp(currentNewName, newNameToUse) != 0)
+                    {
+                        // Send the manager thread the item processed message
+                        PostMessage(pwtd->hwndManager, SRM_REGEX_ITEM_UPDATED, GetCurrentThreadId(), id);
+                    }
+                    CoTaskMemFree(newName);
+                    CoTaskMemFree(currentNewName);
+                    CoTaskMemFree(originalName);
                 }
+                CoTaskMemFree(replaceTerm);
             }
 
             // Send the manager thread the completion message
             PostMessage(pwtd->hwndManager, SRM_REGEX_COMPLETE, GetCurrentThreadId(), 0);
 
             delete pwtd;
+            
         }
         CoUninitialize();
     }
+    catch (...)
+    {
+        MessageBox(NULL, L"RegexWorkerThread failed to execute.\nPlease report the bug to https://aka.ms/powerToysReportBug", L"PowerToys Settings Error", MB_OK);
```

Comment:
Please open an issue for this string to be localized.
Also check if there are other strings in this lib that needs to be localized.

Reply:
Also `"PowerToys Settings Error"` -> `"PowerRename Error"`

Reply:
Can you verify is it's possible to return an error code to the consumer of the lib so that the caller will generate the error string instead of hard-coding into the lib?

Reply:
>Please open an issue for this string to be localized.
Also check if there are other strings in this lib that needs to be localized.
Also "PowerToys Settings Error" -> "PowerRename Error"

Ok

>Can you verify is it's possible to return an error code to the consumer of the lib so that the caller will generate the error string instead of hard-coding into the lib?

I'll check.